### PR TITLE
Expose StreamingQueryManager

### PR DESCRIPTION
**Pull request by suhsteve**
`SparkSession`:
- public StreamingQueryManager Streams()

`StreamingQuery`:
- public string Id
- public string RunId
- public StreamingQueryException Exception()

`StreamingQueryManager`: 
- public IEnumerable<StreamingQuery> Active
- public StreamingQuery Get(string id)
- public void AwaitAnyTermination()
- public void AwaitAnyTermination(long timeoutMs)
- public void ResetTerminated()

#614

### CODE REVIEW

**Comment by imback82** [comment]
@suhsteve Can you check the test failure?

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+            Assert.IsType<string>(sq.Name);
+
+            Assert.IsType<string>(sq.Id);
+
+            Assert.IsType<string>(sq.RunId);
+
+            Assert.IsType<bool>(sq.IsActive());
+
+            Assert.IsType<bool>(sq.AwaitTermination(1000));
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryTests.cs:43]: nit: can we just do 10? No need to wait for a full second?

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryManagerTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            StreamingQuery[] streamingQueries = sqm.Active().ToArray();
+            Assert.Equal(2, streamingQueries.Length);
+
+            Assert.IsType<StreamingQuery>(sqm.Get(id1));
+            Assert.IsType<StreamingQuery>(sqm.Get(id2));
+
+            sqm.ResetTerminated();
+
+            sqm.AwaitAnyTermination(1000);
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryManagerTests.cs:51]: ditto

**Review by imback82 — APPROVED** [review]
LGTM, thanks @suhsteve!

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryTests.cs (hunk 1 above)
Comment (suhsteve) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryTests.cs]: changed.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryManagerTests.cs (hunk 2 above)
Comment (suhsteve) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/StreamingQueryManagerTests.cs]: changed.

**Pull request merged**